Civi/Schema/Traits/ArrayFormatTrait.php

```diff
@@ -66,7 +66,7 @@ public function loadArray(iterable $values, bool $strict = FALSE) {
       if (is_callable([$this, $setter])) {
         $this->{$setter}($value);
       }
-      else {
+      elseif (!$strict || property_exists($this, $field)) {
```

Comment:
Am I misreading this? I feel like it should be `if strict` not `if not strict`

Reply:
From the function's docblock:

> In strict mode, properties are only accepted if they are formally defined on the current class.

So this is saying:
 - If we're ***not*** in strict mode, always set the property, regardless of if it's being dynamically set (this will cause issues on PHP 8.2, but I think is probably necessary for the `WorkflowMessage` functionality for now)
 - Else if we're not in strict mode, only set the property if the property exists.

I do agree this isn't particuarly readable, but I am struggling to see a better way of structuring it at the moment.

Reply:
OK thanks @braders - I'll try to get input from @colemanw. I'm not quite sure why it's needed for `WorkflowMessage` - I think possibly for an approach within `WorkflowMessage` that's not really in use (ie whereever I've used them I've defined the properties)

Reply:
Thanks @eileenmcnaughton. The actual relevant `WorkflowMessage` code is this:

```
$parsed = ReflectionUtils::getCodeDocs($property, 'Property');
$field = new \Civi\WorkflowMessage\FieldSpec();
$field->setName($property->getName())->loadArray($parsed);
```

so the properties are all being set on `FieldSpec`, not on an actual instance of `GenericWorkflowMessage` which would have the properties. (This isn't an area of Civi I'm very familiar with though, so I may be misundertanding what's going on)

Reply:
Hmm @totten might be able to comment then - I don't really understand that code either

FWIW I have been building out the `WorkflowTemplate` classes using only properties with a declared `get` method or that are declared in the annotations. From memory there was also the idea that 'any' property that was on the class might have a pseudo-get on it & I feel like this code might be to support that?

Reply:
Yeah, the `FieldSpec` classes aren't things you'd patch every day. ;)

To clarify what it does, I believe it's for this kind of code:

https://github.com/civicrm/civicrm-core/blob/92400d703b41cec7925e62eb3981ef77c0471b6d/Civi/Api4/Action/Contact/MergeDuplicates.php#L46-L55

https://github.com/civicrm/civicrm-core/blob/8193ed744564d3528c9e4fb440be1a4b8004bfab/CRM/Contribute/WorkflowMessage/ContributionTrait.php#L18-L22

Note the special annotations like `@required`, `@options` and `@scope`. A "field spec" is a list of all those annotations/metadata that you'd put on a field. For example:

* `MergeDuplicates` is an APIv4 action-class. The field `$mode` has a corresponding field-spec (per `Civi\Api4\Service\Spec\FieldSpec`) with info like `$fieldSpec->required` and `$fieldSpec->options`.
* `ContributionInvoiceReceipt` is a workflow-message class. The field `$contributionId` has a corresponding field-spec (per `Civi\WorkflowMessage\FieldSpec`) with info like `$fieldSpec->scope`

The process of loading annotations is roughly:

1. Get the annotations as a big `string` ("@var int\n@required\n@options a,b,c")
2. Convert the `string` to `array` (`["var" => "int", "required" => true, "options" => "a,b,c"]`)
3. Convert the `array` to object properties (`new FieldSpec()` and `$fieldSpec->setRequired($arr['required'])`)

I've focused on annotations here because that's a common data-source which is (comparatively) easy to visualize, but I think there may be other sources of field-specs (e.g. database/extensions/hooks).

(*I haven't read the rest of the PR or got my head into the actual warnings/edge-cases. This comment is 50% explaining and 50% trying-to-remember for myself...*)